corehq/apps/app_manager/static/app_manager/js/forms/form_designer.js

```diff
@@ -54,7 +54,7 @@ hqDefine("app_manager/js/forms/form_designer", function () {
                 $('#formdesigner').removeClass('loading');
                 $('#formdesigner .fd-content-left .fd-head-text').before(
                     $('#fd-hq-edit-formname-button').html()
-                );
+                ).text(initialPageData('form_name'));
```

Comment:
It might be worth putting a comment here and/or in vellum: https://github.com/dimagi/Vellum/blob/master/src/core.js#L1276

Is this guaranteed to happen after the vellum code?

Reply:
This is just outside of what you can see from the diff, but the line above is

```
formLoadedCallback: function () {
```
so my understanding is that yes this is guaranteed to happen after the vellum code that wrote the non-translated form name there initially. I also confirmed in testing that the correct text is shown, and that it is also updated when the form name is edited via the modal.

Reply:
A comment sounds like a good idea. I'll describe the way the top left area is taken over by app builder code to translate the form name and add edit functionality.